File: DSA/pranavdixit_590014418/Day_34_Question_1.c
#include <stdio.h>

int addOne(int n) {
    int mask = 1;
    while (n & mask) {
        n ^= mask;
        mask <<= 1;
    }
    n ^= mask;
    return n;
}

int main() {
    int n = 4;
    int result = addOne(n);
    printf("Result: %d\n", result);
    return 0;
}

File: DSA/VenikaPanwar_590013755/Day_34_Question_1.java
import java.util.Scanner;

public class AddOneBitwise {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number to add 1: ");
        int n = sc.nextInt();

        int mask = 1;

        while ((n & mask) != 0) {
            n = n ^ mask;    // toggle current bit from 1 to 0
            mask <<= 1;      // move to next higher bit
        }

        n = n ^ mask;        // set first 0 bit to 1 (carry ends here)

        System.out.println("Result: " + n);
    }
}
